cats/cats-sql/src/main/kotlin/com/netflix/spinnaker/cats/sql/cluster/SqlClusteredAgentScheduler.kt

```diff
@@ -233,6 +234,16 @@ class SqlClusteredAgentScheduler(
     return trimmedCandidates
   }
 
+  private fun cleanupZombieAgents() {
+    val zombieAgentThreshold = dynamicConfigService.getConfig(Long::class.java, "sql.agent.zombie-threshold-ms", 3600000)
+    activeAgents
+      .filter { it.value.currentTime < System.currentTimeMillis() - zombieAgentThreshold }
+      .forEach {
+        log.warn("Found zombie agent {}, removing it", it.key)
+        activeAgents.remove(it.key, it.value)
```

Comment:
Instead of `activeAgents.remove` let's call `agentCompleted` to make sure the database lock is also released.

Reply:
@german-muzquiz, thanks for the detailed review. If I get it right, I see there is already a mechanism for cleaning expired locks from the db [here](https://github.com/spinnaker/clouddriver/blob/12abd4d778d22829707137673717466e5334511a/cats/cats-sql/src/main/kotlin/com/netflix/spinnaker/cats/sql/cluster/SqlClusteredAgentScheduler.kt#L185). The problem was that the local list is not cleanup never.

Reply:
Ah you're right, it's already cleared when trying to acquire other locks